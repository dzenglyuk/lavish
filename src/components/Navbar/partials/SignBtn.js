import React from "react";
import { Link } from "react-router-dom";

import Button from "../../../components/Button/Main";

const SignBtn = ({ button, handleClick }) => {
    return (
        <li className="nav-btn" onClick={handleClick}>
            <Link to="/sign-up" className="btn-link" data-testid="sign-btn">
                <Button
                    btnStyle="btn--outline"
                    btnSize={button ? null : "btn--mobile"}
                >
                    SIGN UP
                </Button>
            </Link>
        </li>
    );
};

SignBtn.defaultProps = {
    button: false,
    handleClick: null
};

export default React.memo(SignBtn);
