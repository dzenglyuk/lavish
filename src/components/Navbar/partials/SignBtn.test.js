import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { StaticRouter as Router } from "react-router";
import userEvent from "@testing-library/user-event";

import SignBtn from "./SignBtn";

test("Renders correctly", () => {
    const { getByTestId, getByText } = render(
        <Router>
            <SignBtn />
        </Router>
    );

    getByTestId("sign-btn");
    getByText("SIGN UP");
    expect(getByText("SIGN UP").className).toMatch(/btn--mobile/);
    expect(getByTestId("sign-btn")).toHaveProperty(
        "href",
        "http://localhost/sign-up"
    );
});

test("Handles click", () => {
    const handleClick = jest.fn();
    const { getByTestId } = render(
        <Router>
            <SignBtn handleClick={handleClick}/>
        </Router>
    );
    userEvent.click(getByTestId("sign-btn").parentNode);
    expect(handleClick).toHaveBeenCalledTimes(1);
});

test("Matches the snapshot 1", () => {
    const tree = renderer
        .create(
            <Router>
                <SignBtn />
            </Router>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test("Matches the snapshot 2", () => {
    const tree = renderer
        .create(
            <Router>
                <SignBtn button={true} />
            </Router>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
