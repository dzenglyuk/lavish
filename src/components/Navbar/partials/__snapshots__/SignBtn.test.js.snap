// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1 1`] = `
<li
  className="nav-btn"
  onClick={null}
>
  <a
    className="btn-link"
    data-testid="sign-btn"
    href="/sign-up"
    onClick={[Function]}
  >
    <button
      className="btn 
			btn--outline
			btn--mobile
			"
    >
      SIGN UP
    </button>
  </a>
</li>
`;

exports[`Matches the snapshot 2 1`] = `
<li
  className="nav-btn"
  onClick={null}
>
  <a
    className="btn-link"
    data-testid="sign-btn"
    href="/sign-up"
    onClick={[Function]}
  >
    <button
      className="btn 
			btn--outline
			btn--medium
			"
    >
      SIGN UP
    </button>
  </a>
</li>
`;
